File: algorithms/1_easy/P/03146_permutation_difference_between_two_strings/solution.cpp
class Solution {
public:
    int findPermutationDifference(string s, string t) {
        const size_t len{s.size()};
        array<int, 26> position_diff{};

        for (size_t i{0}; i < len; ++i) {
            position_diff[static_cast<size_t>(s[i] - 'a')] -= static_cast<int>(i);
            position_diff[static_cast<size_t>(t[i] - 'a')] += static_cast<int>(i);
        }

        int sum{0};
        for (const int &p: position_diff) {
            sum += abs(p);
        }

        return sum;
    }
};

File: algorithms/1_easy/P/03146_permutation_difference_between_two_strings/solution.py
class Solution:
    def findPermutationDifference(self, s: str, t: str) -> int:
        len_str = len(s)
        position_diff = [0] * 26
        ord_a = ord('a')

        for i in range(len_str):
            position_diff[ord(s[i]) - ord_a] -= i
            position_diff[ord(t[i]) - ord_a] += i

        return sum(abs(n) for n in position_diff)
